app/src/main/java/com/target/dealbrowserpoc/dealbrowser/DealListItemAdapter.kt

```diff
@@ -0,0 +1,60 @@
+package com.target.dealbrowserpoc.dealbrowser
+
+import android.content.Context
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageView
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+import com.squareup.picasso.Picasso
+import com.target.dealbrowserpoc.dealbrowser.service.model.DealItem
+import kotlinx.android.synthetic.main.deal_list_item.view.*
+import kotlinx.android.synthetic.main.fragment_details.*
+
+class DealListItemAdapter(
+        private val context: Context,
+        private val items: List<DealItem>,
+        private val onItemClickListener: OnItemClickListener) : RecyclerView.Adapter<DealListItemAdapter.ViewHolder>() {
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
+        val v = LayoutInflater.from(parent.context)
+                .inflate(R.layout.deal_list_item, parent, false)
+        return ViewHolder(v)
+    }
+
+    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+        val item = getItem(position)
+        holder.itemPrice.text = item.originalPrice
+        holder.title.text = item.title
+        Picasso.with(context).load(item.image).placeholder(R.drawable.image_14741397)
```

Comment:
Picasso can be abstracted out 
Now since it is with in fragment can't be mocked